GameOfLife/GameOfLife/Services/RulesApplier.cs

```diff
@@ -0,0 +1,138 @@
+        public void DetermineCellsDestiny(GameFieldModel gameField, bool disableWrappingAroundField)
+        {
+            for (int i = 0; i < gameField.Length; i++)
+            {
+                for (int j = 0; j < gameField.Width; j++)
+                {
+                    if (gameField.GameField[i, j] == AliveCellSymbol)
+                    {
+                        neighboursCount = CountNeighbourCells(gameField, i, j, false);
+                        switch (disableWrappingAroundField)
+                        {
+                            case false:
+                                if (neighboursCount < 2 || neighboursCount > 3)
+                                {
+                                    _cellsToDie.Add((i, j));
+                                }
+                                break;
+
+                            case true:
+                                if (neighboursCount < 2 || neighboursCount > 3 || i == gameField.Length - 1 || j == gameField.Width - 1)
+                                {
+                                    _cellsToDie.Add((i, j));
+                                }
+                                break;
+                        }
+                    }
+                    else
+                    {
+                        neighboursCount = CountNeighbourCells(gameField, i, j, true);
+                        switch (disableWrappingAroundField)
+                        {
+                            case false:
+                                if (neighboursCount == 3)
+                                {
+                                    _cellsToBeBorn.Add((i, j));
+                                }
+                                break;
+
+                            case true:
+                                if (neighboursCount == 3 && i != gameField.Length - 1 && j != gameField.Width - 1)
+                                {
+                                    _cellsToBeBorn.Add((i, j));
+                                }
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Method to count each cell's alive and dead neighbours to later determine which will die and which will be reborn.
+        /// </summary>
+        /// <param name="gameField">An instance of the GameFieldModel class that stores the game field and its properties.</param>
+        /// <param name="x">Horizontal coordinate of a cell which neighbours are being counted.</param>
+        /// <param name="y">Vertical coordinate of a cell which neighbours are being counted</param>
+        /// <param name="checkDeadCell">Parameter which shows whether the cell, which neighbours are being counted, is alive or dead.</param>
+        /// <returns>Returns the number of cell's neighbours.</returns>
+        private int CountNeighbourCells(GameFieldModel gameField, int x, int y, bool checkDeadCell)
```

Comment:
Parameters x and y should be replaced by descriptive ones.

Reply:
Changed these names.